src/components/Layer/Layer.js

```diff
@@ -1,38 +1,101 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-
-import core from 'core';
 
+import Icon from 'components/Icon';
 import Input from 'components/Input';
 
+import core from 'core';
+
 import './Layer.scss';
 
 class Layer extends React.PureComponent {
   static propTypes = {
     layer: PropTypes.object.isRequired,
-    layers: PropTypes.arrayOf(PropTypes.object).isRequired,
-    index: PropTypes.number.isRequired,
-  }
+  };
 
+  state = {
+    isExpanded: false,
+  };
+
+  unCheckChildren = (layer) => {
+    // new references for redux state
+    const newLayer = {...layer};
+    layer.children && layer.children.forEach((childLayer, i) => {
+      let newChildLayer = {...childLayer};
+      newChildLayer.visible = false;
+      newChildLayer = this.unCheckChildren(newChildLayer);
+      newLayer.children[i] = newChildLayer;
+    });
+    return newLayer;
+  }
 
   onChange = e => {
-    const { index, layers } = this.props;
+    const { updateLayer, layer } = this.props;
```

Comment:
Can you add `updateLayer` to propTypes as well. Thanks!

Reply:
Done